### ISSUE TITLE
Error on connection to AWS Aurora (MySQL)

### ISSUE THREAD

**Issue by vfilimonov** [dbeaver/dbeaver#6922]
#### System information: 
- Operating system (distribution) and version: **OSX 10.14.6**
- DBeaver version: **6.2.1**
- Additional extensions: None

#### Connection specification:
- Database name and version: **AWS RDS Aurora (MySQL)**
- Driver name: **MySQL**
- Do you use tunnels or proxies (SSH, SOCKS, etc)?: **SSH tunnel**

#### Describe the problem you're observing:

Could not connect to the AWS Aurora database (using default driver settings). Each attempt results in an error
```
Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.
```
I tried to increase timeout to 60 seconds - it did not help.

- The database is not paused
- Other clients (e.g. Sequel Pro, Navicat) connect to the same server via SSH tunnel without problems
- The connection is via SSH tunnel, which works fine ("Test tunnel configuration" passes)

I would appreciate any help!

#### Debug log

<details>
<summary>Contents of dbeaver-debug.log (after establishing SSH connection):</summary>
<p>

```
2019-10-05 22:40:27.892 - SSH INFO: Next authentication method: publickey
2019-10-05 22:40:28.017 - SSH INFO: Authentication succeeded (publickey).
2019-10-05 22:40:28.023 - > Start DBeaver MySQL Support [org.jkiss.dbeaver.ext.mysql 2.1.102.201909221630]
2019-10-05 22:40:28.232 - Connection failed (mysql5-16d9d8f0be4-7063b91746ac93d4)
2019-10-05 22:40:28.235 - SSH INFO: Disconnecting from xxx.xxx.xxx.xxx port 22
2019-10-05 22:40:28.238 - SSH INFO: Caught an exception, leaving main loop due to Socket closed
2019-10-05 22:40:28.286 - org.jkiss.dbeaver.model.exec.DBCConnectException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
org.jkiss.dbeaver.model.exec.DBCConnectException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.openConnection(JDBCDataSource.java:179)
	at org.jkiss.dbeaver.ext.mysql.model.MySQLDataSource.openConnection(MySQLDataSource.java:408)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext.connect(JDBCExecutionContext.java:89)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.initializeMainContext(JDBCRemoteInstance.java:88)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.<init>(JDBCRemoteInstance.java:54)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.initializeRemoteInstance(JDBCDataSource.java:105)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.<init>(JDBCDataSource.java:100)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.<init>(JDBCDataSource.java:89)
	at org.jkiss.dbeaver.ext.mysql.model.MySQLDataSource.<init>(MySQLDataSource.java:80)
	at org.jkiss.dbeaver.ext.mysql.MySQLDataSourceProvider.openDataSource(MySQLDataSourceProvider.java:126)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:853)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:70)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.runSync(ConnectJob.java:98)
	at org.jkiss.dbeaver.ui.actions.datasource.DataSourceHandler.connectToDataSource(DataSourceHandler.java:106)
	at org.jkiss.dbeaver.ui.actions.datasource.UIServiceConnectionsImpl.initConnection(UIServiceConnectionsImpl.java:68)
	at org.jkiss.dbeaver.model.navigator.DBNDataSource.initializeNode(DBNDataSource.java:151)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:199)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.model.navigator.DBNUtils.getNodeChildrenFiltered(DBNUtils.java:69)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:86)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:71)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:989)
	at com.mysql.jdbc.MysqlIO.readPacket(MysqlIO.java:632)
	at com.mysql.jdbc.MysqlIO.doHandshake(MysqlIO.java:1016)
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2191)
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2222)
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2017)
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:779)
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:389)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.lambda$0(JDBCDataSource.java:148)
	at org.jkiss.dbeaver.utils.RuntimeUtils$MonitoringTask.run(RuntimeUtils.java:289)
	at org.jkiss.dbeaver.utils.RuntimeUtils$1.run(RuntimeUtils.java:208)
	... 2 more
Caused by: java.io.EOFException: Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.
	at com.mysql.jdbc.MysqlIO.readFully(MysqlIO.java:3011)
	at com.mysql.jdbc.MysqlIO.readPacket(MysqlIO.java:567)
	... 18 more

```
</p>
</details>

**Comment by DemonTPx** [comment]
I'm getting exactly the same error using **DBeaver 6.2.2** on **Linux Mint 19.2** when connecting to two different **AWS RDS Aurora MySQL 5.6** databases. I'm not using an SSH tunnel, but am using SSL

    Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.

Connecting to these databases using other methods (`mysql` from the command line or other mysql clients) just works as usual.

Connecting to any other database also works fine, including a few AWS RDS Aurora MySQL 5.7 instances and various mysql and mariadb servers.

**Comment by maxvisser** [comment]
Just googled this problem; I am facing this issue on Windows. Just while I was typing this comment I saw I entered my QA vps instead of production and therefor used the wrong credentials.

I have a bastion server with a direct connection to an internal database (same network) 
I fill in on General hostname the IP of this internal server. I use AWS; on route53 you can make internal DNS zones. (user friendly names for you servers ;)) I configured a new server. Please make sure you setup everything right on the general tab and test locally on the server if you can connect user (mysql -u username -h host -p)

**Comment by vfilimonov** [comment]
Hello all,

@DemonTPx - right, I'm on AWS RDS Aurora MySQL 5.6.10a - that is the only version which is supported by Aurora Serverless in my region.

@maxvisser -  Since it is a serverless version, there's no access to the instance, but the DB could be accessed from other clients via the same SSH tunnel and using the same configuration.

I've just checked using dbeaver v6.2.4 - the problems still persists there.

**Comment by maxvisser** [comment]
@vfilimonov I also use RDS (two serverless instances mysql 8.0 and 5.6); So our configuration should be the same. Maybe you can show me what you filled in General tab?

**Comment by vfilimonov** [comment]
Hello @maxvisser 

<img width="1329" alt="Screenshot 2019-11-14 at 21 06 28 " src="https://user-images.githubusercontent.com/1324881/68892663-c8a92700-0723-11ea-9e81-60c1ad38d38d.png">

The driver is the one which is default for "MySQL". I suspect that it might cause the issue. Which one do you use for your mysql 5.6?

Connection settings are default except for "connectTimeout" which I set for 60000. The database is not paused when I try to connect.

**Comment by maxvisser** [comment]
I see you did not configure a local client
![image](https://user-images.githubusercontent.com/10818702/69430380-c6535800-0d35-11ea-877e-599d2b16d5c5.png)

**Comment by vfilimonov** [comment]
Thanks for noticing, @maxvisser!
Is it important for dbeaver to have a local client? I thought that the mysql driver should be enough to have (at least all other apps: Navicat, SequelPro, DataGrip work fine without local client)...

Unfortunately installing mysql-client (8.0.18) and setting for dbeaver does not change anything - the error is still the same.

**Comment by jebeaudet** [comment]
Might be unrelated but if you're connected to aurora with tls, they issued a fix yesterday for 5.6 instances https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Updates.1179.html, there were some weird issues when connecting with the now expired CA bundle and even just trying to connect with an unsupported tls version.
